lib/ansible/modules/packaging/language/pip.py

```diff
@@ -499,6 +500,8 @@ class Package:
     test whether a package is already satisfied.
     """
 
+    _CANONICALIZE_RE = re.compile(r'[-_.]+')
```

Comment:
Shouldn't `.` and `-` be escaped properly?

Reply:
Neither of those are special characters in python so they don't have to be escaped for python strings.  Inside of a character class regex characters like ```.``` don't have to be escaped because they don't have special meaning there. ```-``` doesn't have to be escaped in a character class as long as it is the first character in the character class.